hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestNodeDecommissionManager.java

```diff
@@ -56,17 +75,50 @@ public class TestNodeDecommissionManager {
   private NodeDecommissionManager decom;
   private StorageContainerManager scm;
   private NodeManager nodeManager;
+  private ContainerManager containerManager;
   private OzoneConfiguration conf;
+  @TempDir
+  private File testDir;
+  private DBStore dbStore;
+  private SCMHAManager scmhaManager;
+  private SequenceIdGenerator sequenceIdGen;
+  private ContainerReplicaPendingOps pendingOpsMock;
 
   @BeforeEach
   void setup(@TempDir File dir) throws Exception {
     conf = new OzoneConfiguration();
     conf.set(HddsConfigKeys.OZONE_METADATA_DIRS, dir.getAbsolutePath());
-    nodeManager = createNodeManager(conf);
-    decom = new NodeDecommissionManager(conf, nodeManager,
+    scm = HddsTestUtils.getScm(conf);
+    nodeManager = scm.getScmNodeManager();
+    final OzoneConfiguration ozConf = SCMTestUtils.getConf(testDir);
+    dbStore = DBStoreBuilder.createDBStore(
+        ozConf, new SCMDBDefinition());
+    scmhaManager = SCMHAManagerStub.getInstance(true);
+    sequenceIdGen = new SequenceIdGenerator(
+        ozConf, scmhaManager, SCMDBDefinition.SEQUENCE_ID.getTable(dbStore));
+    final PipelineManager pipelineManager =
+        new MockPipelineManager(dbStore, scmhaManager, nodeManager);
+    pipelineManager.createPipeline(RatisReplicationConfig.getInstance(
+        HddsProtos.ReplicationFactor.THREE));
+    pendingOpsMock = mock(ContainerReplicaPendingOps.class);
+    containerManager = new ContainerManagerImpl(ozConf,
```

Comment:
I have not checked the test cases yet, but is it necessary to have the actual container manager implementation here? Can we not mock it instead? If we mock it, this test doesn't need to depend on the actual implementation.

Reply:
Thanks for the review, I have used mock for these tests now